internal/output/vertical.go

```diff
@@ -11,30 +11,91 @@ import (
 )
 
 func PrintVerticalResults(vulnResult *models.VulnerabilityResults, outputWriter io.Writer) {
-	for i, result := range vulnResult.Results {
-		printVerticalHeader(result, outputWriter)
-		printVerticalVulnerabilities(result, outputWriter)
+	// Add a newline to separate results from logs.
+	fmt.Fprintln(outputWriter)
+	outputResult := BuildResults(vulnResult)
+	printSummary(outputResult, outputWriter)
+	if outputResult.IsContainerScanning {
+		printBaseImages(outputResult.ImageInfo, outputWriter)
+	}
 
-		if len(vulnResult.ExperimentalAnalysisConfig.Licenses.Allowlist) > 0 {
-			printVerticalLicenseViolations(result, outputWriter)
+	for i, ecosystem := range outputResult.Ecosystems {
+		fmt.Fprintf(outputWriter, "%s", text.FgGreen.Sprintf("%s\n\n", ecosystem.Name))
+		for j, source := range ecosystem.Sources {
+			printVerticalHeader(source, outputWriter)
+			printVerticalVulnerabilities(source, outputResult.IsContainerScanning, outputWriter)
+			if j < len(ecosystem.Sources)-1 {
+				fmt.Fprintln(outputWriter)
+			}
 		}
 
-		if i < len(vulnResult.Results)-1 {
+		if i < len(outputResult.Ecosystems)-1 {
 			fmt.Fprintln(outputWriter)
 		}
 	}
+
+	fmt.Fprintln(outputWriter)
+}
+
+func printSummary(result Result, out io.Writer) {
+	summary := fmt.Sprintf(
+		"Total %[1]d packages affected by %[2]d vulnerabilities (%[3]s, %[4]s, %[5]s, %[6]s, %[7]s) from %[8]s.\n"+
+			"%[9]d vulnerabilities have fixes available.\n",
+		result.PackageTypeCount.Regular,
+		result.VulnTypeSummary.All,
+		text.FgRed.Sprintf("%d Critical", result.VulnCount.SeverityCount.Critical),
+		text.FgHiYellow.Sprintf("%d High", result.VulnCount.SeverityCount.High),
+		text.FgYellow.Sprintf("%d Medium", result.VulnCount.SeverityCount.Medium),
+		text.FgHiCyan.Sprintf("%d Low", result.VulnCount.SeverityCount.Low),
+		text.FgCyan.Sprintf("%d Unknown", result.VulnCount.SeverityCount.Unknown),
+		text.FgGreen.Sprintf("%d ecosystems", len(result.Ecosystems)),
+		result.VulnCount.FixableCount.Fixed,
+	)
+	fmt.Fprintln(out, summary)
+}
+
+func printBaseImages(imageResult ImageInfo, out io.Writer) {
+	fmt.Fprintf(out, "Container image information:\n")
+	fmt.Fprintf(out, "  OS version: %s\n", text.FgGreen.Sprintf("%s", imageResult.OS))
+	for _, baseImage := range imageResult.AllBaseImages {
+		baseImageString := text.FgYellow.Sprintf("Base Image %d (%s)", baseImage.Index, getBaseImageName(baseImage))
+		if baseImage.Index == 0 {
+			baseImageString = text.FgYellow.Sprintf("Your Image")
+		}
+		fmt.Fprintf(out, "  %s:\n", baseImageString)
+		for _, layer := range baseImage.AllLayers {
+			layerCommand := formatLayerCommand(layer.LayerMetadata.Command)[0]
+			layerCommand = truncate(layerCommand, 100)
+			fmt.Fprintf(out, "    %s %s", text.FgCyan.Sprintf("Layer %d", layer.Index), text.Italic.Sprintf("%s", layerCommand))
```

Comment:
nit: I feel like it would look slightly nicer if the numbers were padded to the largest length, so all the text aligned

i.e. in your screenshot:

```
Layer 8  /bin/sh ...
Layer 9  /bin/sh
Layer 10 /bin/sh ...
```

that makes my brain happy, and I don't think it'd be very expensive to calculate

Reply:
I didn't notice this. Thanks for pointing out. Added some padding based on the max number of digits